lib/misty/http/net_http.rb

```diff
@@ -1,14 +1,20 @@
 module Misty
   module HTTP
+    # This class implements the http request routine.
     module NetHTTP
-      def self.net_http(endpoint, ssl_verify_mode, log)
-        http = Net::HTTP.new(endpoint.host, endpoint.port)
-        http.set_debug_output(log) if log.level == Logger::DEBUG
-        if endpoint.scheme == "https"
-          http.use_ssl = true
-          http.verify_mode = OpenSSL::SSL::VERIFY_NONE unless ssl_verify_mode
+      def self.http_request(uri, options = {})
+        verify_mode = if options[:ssl_verify_mode]
+                        OpenSSL::SSL::VERIFY_PEER
+                      else
+                        OpenSSL::SSL::VERIFY_NONE
```

Comment:
Although this makes it more explicit meanwhile this is not needed since VERIFY_NONE is the default.